src/core/ddsi/src/ddsi_serdata_default.c

```diff
@@ -452,6 +452,23 @@ static struct ddsi_serdata* serdata_default_from_iox(const struct ddsi_sertype*
 
   return (struct ddsi_serdata*)d;
 }
+
+// Creates a serdata for the case where only iceoryx is required (i.e. no network).
+// This skips expensive serialization and just takes ownership of the iceoryx buffer.
+// Computing the keyhash is currently still required.
+static struct ddsi_serdata *ddsi_serdata_default_from_sample_for_iox (const struct ddsi_sertype *type, enum ddsi_serdata_kind kind, const char *sample) {
+  const struct ddsi_sertype_default *t = (const struct ddsi_sertype_default *)type;
+  struct ddsi_serdata_default *d = serdata_default_new(t, kind);
+
+  if(d == NULL) 
+    return NULL;
+
+  gen_keyhash_from_sample (t, &d->keyhash, sample);
+
+  struct ddsi_serdata *serdata = (struct ddsi_serdata*) d;
+  serdata->iox_chunk = SHIFT_BACK_TO_ICEORYX_HEADER(sample);
```

Comment:
This assumes that sample is preceeded by an iceoryx header (which is the case currently as far as I can see). If this assumption is broken in the future it will cause memory access problems.